src/pages/_app.tsx

```diff
@@ -33,34 +33,10 @@ if (
 const MyApp = ({ pageProps, Component }: AppProps) => {
   const router = useRouter();
 
-  const routerReplace = (url: string) => {
-    /**
-     * Apologies for this line of code. The reason this is here
-     * is that XState + React Fast Refresh causes an error:
-     *
-     * Error: Unable to send event to child 'ctx => ctx.sourceRef'
-     * from service 'auth'.
-     *
-     * router.replace causes this in development, but not in prod
-     *
-     * So, we use window.location.href in development (with the /viz
-     * prefix which Next won't automatically add) and router.replace in prod
```

Comment:
After this refactor I couldn't find any problems during development and also before my latest fix ( https://github.com/statelyai/xstate-viz/pull/245 ) this was an issue in production. The state of the world/codebase could have changed in the meantime and I think this is no longer necessary.

Reply:
Very nice!